### PR TITLE
Support TTC v2

### LINKED ISSUE #11: Fix TTC parsing
```
I have no idea how to submit patches, so forgive me if this is an inappropriate 
venue.

Attached is a patch fixing TTC parsing.
The constant 12 was hardcoded instead of using `offset`.

Tested on a TTC v2.0 file, I don't have a v1.0 file to test on but based on 
what I understand of the file format I don't think there should be any issues 
with that.
```

Original issue reported on code.google.com by `magician...@gmail.com` on 10 Sep 2014 at 4:30

Attachments:
- [freetype-go.patch](https://storage.googleapis.com/google-code-attachments/freetype-go/issue-11/comment-0/freetype-go.patch)

### PR DESCRIPTION
**Pull request by mattn**
added test for parsing TTC but I don't include testdata.

Close #11

### CODE REVIEW

File: truetype/truetype_test.go [hunk 1]
```diff
@@ -110,7 +110,18 @@ func TestParse(t *testing.T) {
...
+			'!':      4,
+			'A':      36,
+			'V':      57,
+			'É':      101,
+			'ﬂ':      193,
+			'\u22c5': 385,
+			'中':      0,
+		},
+
+		"x-NotoSansCJK-Black.ttc": {
```
Comment (nigeltao) [inline truetype/truetype_test.go]: If you're going to add an "x-foo" font, you need to update the testdata/make-other-hinting-txts.sh script.

File: truetype/truetype_test.go [hunk 2]
```diff
@@ -110,7 +110,18 @@ func TestParse(t *testing.T) {
...
 			' ':      3,
```
Comment (nigeltao) [inline truetype/truetype_test.go]: These copy/pasted numbers aren't right for NotoSansCJK-Black.ttc.

File: truetype/truetype_test.go [hunk 3]
```diff
@@ -124,7 +135,7 @@ func TestIndex(t *testing.T) {
 		// The x-etc test cases use those versions of the .ttf files provided
```
Comment (nigeltao) [inline truetype/truetype_test.go]: This comment needs updating.

File: truetype/truetype.go [hunk 4]
```diff
@@ -550,7 +550,7 @@ func parse(ttf []byte, offset int) (font *Font, err error) {
 			return
 		}
 		ttcVersion, offset := u32(ttf, offset), offset+4
-		if ttcVersion != 0x00010000 {
+		if ttcVersion != 0x00010000 && ttcVersion != 0x00020000 {
 			// TODO: support TTC version 2.0, once I have such a .ttc file to test with.
```
Comment (nigeltao) [inline truetype/truetype.go]: Delete this line.

File: truetype/truetype.go [hunk 5]
```diff
@@ -578,14 +578,16 @@ func parse(ttf []byte, offset int) (font *Font, err error) {
 		return
 	}
 	n, offset := int(u16(ttf, offset)), offset+2
-	if len(ttf) < 16*n+12 {
+	offset = offset + 2*3 // skip 3 short fields
```
Comment (nigeltao) [inline truetype/truetype.go]: Change this line to:
offset += 6 // Skip the searchRange, entrySelector and rangeShift.

**Comment by mattn** [comment]
Sorry, NotoSansCJK-Black is not good for this test because this contains OTTO (Adobe OpenType OTTO). So it cause parse error. I'll try to find font for this.

**Comment by mattn** [comment]
updated. I copied TestTTC.ttc from fonttools project.

https://github.com/fonttools/fonttools/tree/master/Tests

**Comment by nigeltao** [comment]
I'm still not convinced that adding a .ttc file to testdata/ is really worth it. I'd just leave it out.

**Comment by mattn** [comment]
Do you prefer it is x-foo?

**Comment by nigeltao** [comment]
I prefer to leave it out.

**Comment by mattn** [comment]
you mean it's okay without test, right?

**Comment by nigeltao** [comment]
Yes, OK without test.

**Comment by mattn** [comment]
please take a look.

**Comment by nigeltao** [comment]
Thanks for your patience.

**Pull request merged**